test/CarouselSpec.js

```diff
@@ -65,7 +65,38 @@ describe('<Carousel>', () => {
     wrapper.find('.carousel-indicators li').first().simulate('click');
   });
 
-  ['onSlide', 'onSlid'].forEach((eventName) => {
+  describe(`ref testing`, () => {
+    let clock;
+
+    beforeEach(() => {
+      clock = sinon.useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+    it(`should allow refs to be attached and expose next, prev functions`, () => {
+      let inputRef = React.createRef();
+      const onSelectSpy = sinon.spy();
+      const wrapper = mount(
+        <Carousel ref={inputRef} onSelect={onSelectSpy} defaultActiveIndex={1}>
+          {items}
+        </Carousel>,
+      );
+      expect(inputRef.current).to.have.property('next');
+      expect(inputRef.current).to.have.property('prev');
+      expect(inputRef.current).to.have.property('element');
+      inputRef.current.next();
+      clock.tick(50);
+      expect(onSelectSpy).to.have.been.calledOnce;
+      inputRef.current.prev();
+      clock.tick(50);
+      expect(onSelectSpy).to.have.been.calledTwice;
+      wrapper.unmount();
+    });
+  });
+
+  [('onSlide', 'onSlid')].forEach((eventName) => {
```

Comment:
this wasn't valid syntax

```suggestion
  ['onSlide', 'onSlid'].forEach((eventName) => {
```